pkg/cmd/admin/router/router.go

```diff
@@ -642,12 +654,12 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 		if cfg.SecretsAsEnv {
 			env.Add(app.Environment{"DEFAULT_CERTIFICATE": string(defaultCert)})
 		} else {
-			// TODO: make --credentials create secrets and bypass service account
 			env.Add(app.Environment{"DEFAULT_CERTIFICATE_PATH": defaultCertificatePath})
 		}
 	}
-
-	secrets, volumes, mounts, err := generateSecretsConfig(cfg, kClient, namespace, defaultCert)
+	env.Add(app.Environment{"DEFAULT_CERTIFICATE_DIR": defaultCertificateDir})
```

Comment:
@smarterclayton When --default-cert is used the admin passes a pem and _PATH points to the tls.crt file in the mounted secret. When a pem is not provided we use a service annotation to create a secret containing the default crt/key in two separate files (not a pem in a single file). In this case the files need to be combined into a pem. _DIR points to the root directory (/etc/pki/private).

You specify _DIR when you have a secret in crt/key (in two files not a pem in the crt) format. 
In this case  _PATH is empty.